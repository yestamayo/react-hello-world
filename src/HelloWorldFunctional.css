.HelloWorldFunctional {
  border: 2px solid black;
  text-align: center;
  background: #CCC;
  color: black;
  margin: 20px;
  padding: 20px;
}