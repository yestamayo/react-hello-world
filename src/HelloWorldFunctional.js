import React, {Component} from 'react';
import './HelloWorldFunctional.css';

class HelloWorldFunctional extends Component{

    constructor(props){
        super(props);
        this.state = { 
            greeting: 'Hola' 
        };
        this.frenchify = this.frenchify.bind(this);
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    frenchify() {
        this.setState({ greeting: 'Bonjour' });
    }

    render() {
        return (
            <div className="HelloWorldFunctional">
                {this.state.greeting} {this.props.name}!
                <br/>
                <button className="frenchify" onClick={this.frenchify}>Frenchify!</button>
                <br/>
                <button className="remove" onClick={this.removeGreeting}>Remove Me!</button>
            </div>
        );
    }

    removeGreeting() {
        this.props.removeGreeting(this.props.name);
    }
}

export default HelloWorldFunctional;
