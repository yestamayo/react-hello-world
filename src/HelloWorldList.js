import React, {Component} from 'react';
import HelloWorldFunctional from './HelloWorldFunctional';
import './HelloWorldList.css';
import AddGreeter from './AddGreeter';

class HelloWorldList extends Component {

    constructor(props){
        super(props);
        this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
        this.addGreeting = this.addGreeting.bind(this);
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    render() {
        return (
            <div className="HelloWorldList">
                <AddGreeter addGreeting={this.addGreeting}/>
                {this.renderGreetings()}
            </div>
        );
    }

    renderGreetings() {
        return this.state.greetings.map(name => (
            <HelloWorldFunctional
            key={name}
            name={name}
            removeGreeting={this.removeGreeting}
            />
        ));
    }

    addGreeting(newName) {
        this.setState({ greetings: [...this.state.greetings, newName] });
    }

    removeGreeting(removeName) {
        const filteredGreetings = this.state.greetings.filter(name => {
            return name !== removeName;
        });
        this.setState({ greetings: filteredGreetings });
    }
}

export default HelloWorldList;