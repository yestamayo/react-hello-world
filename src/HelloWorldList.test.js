import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';
import HelloWorldFunctional from './HelloWorldFunctional';

describe(HelloWorldList, () => {
  const component = shallow(
    <HelloWorldList />
  );
  // Add the rest of our tests here

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorldList />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains an AddGreeter subcomponent', () => {
        expect(component.find(AddGreeter)).toHaveLength(1);
    });

    it('contains the same number of HelloWorldFunctional components as greetings', () => {
        const helloWorlds = component.find(HelloWorldFunctional).length;
        const greetings   = component.state('greetings').length;
        expect(helloWorlds).toEqual(greetings);
    });

    it('adds another greeting when the add greeting function is called', () => {
        const before = component.find(HelloWorldFunctional).length;
        component.instance().addGreeting('Sample');
        const after = component.find(HelloWorldFunctional).length;
        expect(after).toBeGreaterThan(before);
    });

    it('removes a greeting from the list when the remove greeting function is called', () => {
        const before   = component.find(HelloWorldFunctional).length;
        const removeMe = component.state('greetings')[0];
        component.instance().removeGreeting(removeMe);
        const after = component.find(HelloWorldFunctional).length;
        expect(after).toBeLessThan(before);
    });

});