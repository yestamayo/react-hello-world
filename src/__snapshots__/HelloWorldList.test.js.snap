// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {

    constructor(props) {
        super(props);
        this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
        this.addGreeting = this.addGreeting.bind(this);
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    render() {
        return (
            _react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this },
                _react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this }),
                this.renderGreetings()));


    }

    renderGreetings() {
        return this.state.greetings.map(name =>
        _react2.default.createElement(_HelloWorldFunctional2.default, {
            key: name,
            name: name,
            removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }));


    }

    addGreeting(newName) {
        this.setState({ greetings: [...this.state.greetings, newName] });
    }

    removeGreeting(removeName) {
        const filteredGreetings = this.state.greetings.filter(name => {
            return name !== removeName;
        });
        this.setState({ greetings: filteredGreetings });
    }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="HelloWorldFunctional"
  >
    Hola
     
    Jim
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorldFunctional"
  >
    Hola
     
    Sally
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorldFunctional"
  >
    Hola
     
    Bender
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Me!
    </button>
  </div>
</div>
`;
